### PR TITLE
Simplify how data is passed to Long term graph

### LINKED ISSUE #2372: Long-Term Data -> Graphics are fading in an disappear
### Versions
  Pi-hole version is v5.12.2 (Latest: v5.12.2)
  AdminLTE version is v5.15.1 (Latest: v5.15.1)
  FTL version is v5.18.1 (Latest: v5.18.1)

### Platform
- OS and version:  Ubuntu 22.04
- Platform:  LXC on Debian11 (Proxmox)

### Expected behavior

A beautiful graph is visible

### Actual behavior / bug

A beautiful graph is not visible

### Steps to reproduce

1. Go to 'Long-term Data'
2. Click on 'Graphics'
3. Select time range 'today'
4. See error (video added)

## Debug Token

- URL:  https://tricorder.pi-hole.net/G1Ri5hJ7/

### Screenshots

[Bildschirmaufzeichnung vom 23.09.2022, 09:49:03.webm](https://user-images.githubusercontent.com/2614431/191915271-2653092a-318c-41b6-a029-67b2a90e20c8.webm)

### Additional context
* the dashboard graphs are shown correctly
* installed and reinstall with the offical install.sh script on vanilla ubuntu lxc container
* tried firefox and google-chrome
* tried refreshing/deleting cache

Any ideas? 

Kind regards
Henning

### PR DESCRIPTION
**Pull request by rdwebdesign**
### What does this PR aim to accomplish?

- Simplify how data is passed to Long term graph

### How does this PR accomplish the above?

- Remove unnecessary step used to fill all empty time-slots with zeros.
- Set the `min` and `max` for the x-axis on javascript code and let `chartsj` handle the empty slots. 

### What documentation changes (if any) are needed to support this PR?

none

---

**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
- [x] I have read the above and my PR is ready for review. _Check this box to confirm_

### CODE REVIEW

**Comment by PromoFaux** [comment]
Discussion here: https://github.com/pi-hole/AdminLTE/issues/2372

**Comment by rdwebdesign** [comment]
Note:
I rebased the branch.

File: scripts/pi-hole/js/db_graph.js [hunk 1]
```diff
@@ -196,6 +196,8 @@ function updateQueriesOverTime() {
         }
       }
 
+      timeLineChart.options.scales.xAxes[0].ticks.min = from * 1000;
```
Comment (yubiuser) [inline scripts/pi-hole/js/db_graph.js:199]: Why do we need to multiply by 1000 here?

**Review by yubiuser — COMMENTED** [review]
Works with my fake data.

**Comment by yubiuser** [comment]
Confirmed working in https://github.com/pi-hole/AdminLTE/issues/2372

File: scripts/pi-hole/js/db_graph.js (hunk 1 above)
Comment (rdwebdesign) [reply scripts/pi-hole/js/db_graph.js:199]: Because timestamps are in seconds, but javascript uses microseconds.

Comment (yubiuser) [reply scripts/pi-hole/js/db_graph.js:199]: Thanks.

**Review by yubiuser — APPROVED** [review]

**Comment by yubiuser** [comment]
We might need to revise
> Remove unnecessary step used to fill all empty time-slots with zeros.

as `spanGaps: false` isn't going to work anymore. 

Screenshot at 2022-09-29 09-14-08

Instead of adding `0` we could add `NaN` and get a real break like here: https://jsfiddle.net/srq8hzw2/

**Comment by yubiuser** [comment]
Setting it to `NaN`

Screenshot at 2022-09-29 09-25-42

**Comment by yubiuser** [comment]
The code will need adjustment - it now prints nothing if not both datasets (blocked, permitted) contain data. If only one contains data, the other isn't shown for that interval.

**Comment by rdwebdesign** [comment]
I think I fixed.

Please, test the new branch.
If it works, I'll open a PR.

**Pull request merged**